webapp/src/app/ui/avatar/avatar.stories.ts

```diff
@@ -1,95 +1,34 @@
-import { argsToTemplate, type Meta, type StoryObj } from '@storybook/angular';
-import { AppAvatarComponent, args, argTypes } from './avatar.component';
+import { moduleMetadata, type Meta, type StoryObj } from '@storybook/angular';
+import { AvatarComponent } from './avatar.component';
+import { AvatarImageComponent } from './avatar-image.component';
+import { AvatarFallbackComponent } from './avatar-fallback.component';
 
-const meta: Meta<AppAvatarComponent> = {
+const meta: Meta<AvatarComponent> = {
   title: 'UI/Avatar',
-  component: AppAvatarComponent,
-  tags: ['autodocs'],
-  args: {
-    ...args
+  component: AvatarComponent,
+  subcomponents: {
+    AvatarImageComponent,
+    AvatarFallbackComponent
   },
-  argTypes: {
-    ...argTypes
-  }
+  decorators: [
+    moduleMetadata({
+      imports: [AvatarImageComponent, AvatarFallbackComponent]
+    })
+  ],
+  tags: ['autodocs']
 };
 
 export default meta;
-type Story = StoryObj<AppAvatarComponent>;
+type Story = StoryObj<AvatarComponent>;
 
 export const Default: Story = {
-  args: {
-    src: 'https://i.pravatar.cc/40?img=1',
-    alt: 'avatar',
-    class: ''
-  },
-
-  render: (args) => ({
-    props: args,
-    template: `<app-avatar ${argsToTemplate(args)}></app-avatar>`
-  })
-};
-
-export const Small: Story = {
-  args: {
-    size: 'sm',
-    src: 'https://i.pravatar.cc/24?img=1'
-  },
-
-  render: (args) => ({
-    props: args,
-    template: `<app-avatar ${argsToTemplate(args)}></app-avatar>`
-  })
-};
-
-export const Medium: Story = {
-  args: {
-    size: 'default',
-    src: 'https://i.pravatar.cc/40?img=1'
-  },
-
-  render: (args) => ({
-    props: args,
-    template: `<app-avatar ${argsToTemplate(args)}></app-avatar>`
-  })
-};
-
-export const Large: Story = {
-  args: {
-    size: 'lg',
-    src: 'https://i.pravatar.cc/56?img=1',
-    alt: 'avatar',
-    class: ''
-  },
-
-  render: (args) => ({
-    props: args,
-    template: `<app-avatar ${argsToTemplate(args)}></app-avatar>`
-  })
-};
-
-export const WithRandomImage: Story = {
-  args: {
-    size: 'lg',
-    src: 'https://i.pravatar.cc/56',
-    alt: 'avatar'
-  },
-
-  render: (args) => ({
-    props: args,
-    template: `<app-avatar ${argsToTemplate(args)}></app-avatar>`
-  })
-};
-
-export const WithFallback: Story = {
-  args: {
-    size: 'default',
-    src: 'foobar.jpg',
-    fallback: 'https://placehold.co/40',
-    alt: 'fallback'
-  },
-
   render: (args) => ({
     props: args,
-    template: `<app-avatar ${argsToTemplate(args)}></app-avatar>`
+    template: `
+      <app-avatar>
+        <app-avatar-image src="https://github.com/shadcn.png" alt="@shadcn" />
+        <app-avatar-fallback>CN</app-avatar-fallback>
+      </app-avatar>
+    `
```

Comment:
Just for testing purposes it would be great to be able to test different avatars or the fallback component with the interactive controls in Storybook (being able to change the src attribute would suffice I guess).

![image](https://github.com/user-attachments/assets/acb86c1d-5cbb-4e2a-b7a2-f1cafc2d0381)

Reply:
I had some difficulties with that but I can try again, it assumes the props of `AvatarComponent`

Reply:
I guess this works:
https://storybook.js.org/docs/writing-stories/args#args-can-modify-any-aspect-of-your-component

Reply:
Done :)